src/models/data.h

```diff
@@ -468,6 +468,18 @@ struct TopDownResults
     static TopDownResults fromBottomUp(const Data::BottomUpResults& bottomUpData);
 };
 
+struct FrequencyData
+{
+    quint64 time;
+    qreal cost;
+};
+
+struct FrequencyResults
+{
+    QVector<QVector<FrequencyData>> data;
+    QVector<QString> name;
```

Comment:
can you please document what kind of names these are? you may have set that in the past, but I can't come up with a clear idea what we need to name when we are looking at frequency data.

are these per-cpu frequencies, and the names here are actually cpuCoreNames or something like that? and then `data` is something like `perCoreData` ? Better yet, we could make this:

```
struct PerCoreFrequencyData
{
    QString coreName;
    QVector<FrequencyData> frequencies;
};
struct FrequencyResults
{
    QVector<PerCoreFrequencyData> cores;
};
```

if you do that, don't forget the movable type for the new struct

Reply:
Looking below, the data here is per cost, but you must split it per core too - otherwise you'll get _really_ funky values for multi threaded applications :D

but this just shows how confusing these types are as-is, please introduce the per core type as indicated above and then also add the per cost data. in a similar fashion

Reply:
RequencyResults is per cost type so i would suggest the other way around
```
struct FrequencyPerCost
{
    QString name;
    QVector<FrequencyData> cores;
};

struct FrequencyResults
{
    QVector<FrequencyPerCost>
};
```